common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -683,50 +683,64 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
-                state={
-                    'done': user_state.state.get('done'),
-                    'correct_map': user_state.state.get('correct_map'),
-                    'student_answers': user_state.state.get('student_answers'),
-                    'has_saved_answers': user_state.state.get('has_saved_answers'),
-                    'input_state': user_state.state.get('input_state'),
-                    'seed': user_state.state.get('seed'),
-                },
-                seed=user_state.state.get('seed'),
-                # extract_tree=False allows us to work without a fully initialized CapaModule
-                # We'll still be able to find particular data in the XML when we need it
-                extract_tree=False,
-            )
-
-            for answer_id, orig_answers in lcp.student_answers.items():
-                # Some types of problems have data in lcp.student_answers that isn't in lcp.problem_data.
-                # E.g. formulae do this to store the MathML version of the answer.
-                # We exclude these rows from the report because we only need the text-only answer.
-                if answer_id.endswith('_dynamath'):
-                    continue
-
-                if limit_responses and count >= limit_responses:
-                    # End the iterator here
-                    return
-
-                question_text = lcp.find_question_label(answer_id)
-                answer_text = lcp.find_answer_text(answer_id, current_answer=orig_answers)
-                correct_answer_text = lcp.find_correct_answer_text(answer_id)
-
-                count += 1
+                for answer_id, orig_answers in lcp.student_answers.items():
+                    # Some types of problems have data in lcp.student_answers that isn't in lcp.problem_data.
+                    # E.g. formulae do this to store the MathML version of the answer.
+                    # We exclude these rows from the report because we only need the text-only answer.
+                    if answer_id.endswith('_dynamath'):
+                        continue
+
+                    if limit_responses and count >= limit_responses:
+                        # End the iterator here
+                        return
+
+                    question_text = lcp.find_question_label(answer_id)
+                    answer_text = lcp.find_answer_text(answer_id, current_answer=orig_answers)
+                    correct_answer_text = lcp.find_correct_answer_text(answer_id)
+
+                    count += 1
+                    report = {
+                        _("Answer ID"): answer_id,
+                        _("Question"): question_text,
+                        _("Answer"): answer_text,
+                    }
+                    if correct_answer_text is not None:
+                        report[_("Correct Answer")] = correct_answer_text
+                    yield (user_state.username, report)
+            except LoncapaProblemError:
```

Comment:
Will this definitely catch the errors we're seeing. No need to more broadly catch an `Exception` here?

Reply:
Yes, Julia's previous PR throws all errors at codejail level as a LoncapaProblemError.

Reply:
Note that this is also specific to Loncapa problems, not any kind of problem, as each problem type (xblock) implements generate_report_data